### ISSUE TITLE
From the license it's not clear if we can port it on rust

### ISSUE THREAD

**Issue by teamday** [WinampDesktop/winamp#220]

**Comment by jakubtalich** [comment]
I would like to port it to Dart with Flutter UI :-)

**Comment by cvyl** [comment]
True, also not stating if we can port it to brainf*ck either.

**Comment by incycledream** [comment]
I'm gonna rewrite WinAMP in Nix

**Comment by YarosMallorca** [comment]
> I would like to port it to Dart with Flutter UI :-)

That would be awesome, I'm in!

**Comment by ErdbeerbaerLP** [comment]
Llama


> MIT License
> 
> Copyright (c) 2024 ErdbeerbaerLP
> 
> Permission is hereby granted, free of charge, to any person reading this comment, to deal with the comment without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the comment, and to permit persons to whom the comment is furnished to do so, subject to the following conditions:
> 
> The above copyright notice and this permission notice shall be included in all copies or substantial portions of the comment.
> 
> THE COMMENT IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE COMMENT OR THE USE OR OTHER DEALINGS IN THE COMMENT.

**Comment by raynet** [comment]
I think it is super clear, you can, but you cannot distributed the Rust version source code or binary

**Comment by YarosMallorca** [comment]
@raynet so what, we can't make a fork, port it to rust and maintain that project on GitHub?

**Comment by NJdevPro** [comment]
A port to another language amounts to creating a clone. AFAIK, as long as there is no common code, and there is no infringement of copyrighted names, no copyright prevents doing that, what you are doing is effectively another product.

**Comment by bphd** [comment]
> A port to another language amounts to creating a clone. AFAIK, as long as there is no common code, and there is no infringement of copyrighted names, no copyright prevents doing that, what you are doing is effectively another product.

The loophole stable diffusion and LLM use to copy everything without conseuences

**Comment by ARKye03** [comment]
It's time for Tauri lads

**Comment by overanalytcl** [comment]
Also not stating if I am allowed to port Winamp to Delphi either 🤔 which means it doesn't _disallow_ me

**Comment by raynet** [comment]
> @raynet so what, we can't make a fork, port it to rust and maintain that project on GitHub?

Yes, the license doesn'y allow distribution of modified source code, so you can fork it on Github (right guaranteed by Github), make the repo private and port it to Rust there and then use it privately.

**Comment by teamday** [comment]
> Also not stating if I am allowed to port Winamp to Delphi either 🤔 which means it doesn't _disallow_ me

which version of Delphi do you suggest?

**Comment by raynet** [comment]
> Also not stating if I am allowed to port Winamp to Delphi either 🤔 which means it doesn't _disallow_ me

Sadly the copyright law by default doesn't give you any rights to copy and distribute or share the copyrighted files. And licenses specifically prohibits distributing modified source code, so it doesn't matter what port you try to do, or even just fix a typo, any modified fork has to be kept private and not shared,copied or distributed in any way.

**Comment by NJdevPro** [comment]
> > Also not stating if I am allowed to port Winamp to Delphi either 🤔 which means it doesn't _disallow_ me
> 
> Sadly the copyright law by default doesn't give you any rights to copy and distribute or share the copyrighted files. And licenses specifically prohibits distributing modified source code, so it doesn't matter what port you try to do, or even just fix a typo, any modified fork has to be kept private and not shared,copied or distributed in any way.

As long as you don't copy any part of the original code and never reference Winamp, you can totally redistribute your rewrite.
You may not be allowed to distribute the graphics, so you would have to use another skin. Apart from that, you can totally rewrite it in another language and distribute the result, as long as all you say "inspired by a famous player" and not "Winamp", you are legally fine.

**Comment by raynet** [comment]
> As long as you don't copy any part of the original code and never reference Winamp, you can totally redistribute your rewrite. You may not be allowed to distribute the graphics, so you would have to use another skin. Apart from that, you can totally rewrite it in another language and distribute the result, as long as all you say "inspired by a famous player" and not "Winamp", you are legally fine.

I hope you are not giving legal advice, rewriting after forking the repo can be problematic legally, it can be seen as derivate work, thus could be covered by Winamp copyrights. Also you cannot include any of the texts etc that are copyrightable. To be safe and to make <insert language here> rewrite of Winamp, someone should fork Winamp, create detailed software requirement/specification document and new app is written on top of that with devs who haven't seen the Winamp source code.
And this is not legal advice, I am not a lawyer, and live in EU, but have worked with software and US IP laws (via guidance from real lawyers) for 30+ years.
